src/smartcontracts/typesystem/typeExpressionParser.ts

```diff
@@ -26,6 +26,6 @@ export class TypeExpressionParser {
 
     private typeFormulaToType(typeFormula: TypeFormula): Type {
         const typeParameters = typeFormula.typeParameters.map((typeFormula) => this.typeFormulaToType(typeFormula));
-        return new Type(typeFormula.name, typeParameters);
+        return new Type(typeFormula.name, typeParameters, undefined, typeFormula.metadata);
```

Comment:
For double-checking: before, at runtime, I think the second parameter was `TypeCardinality.fixed(1)`, and now it's `undefined`.